playgame.py

```diff
@@ -16,36 +17,45 @@
 logging.basicConfig()
 logger = logging.getLogger(__name__)
 
+class InvalidHanabiMoveException(Exception):
+    def __init__(self, message, move, player_id):
+        self.message = message
+        self.move = move
+        self.player_id = player_id
+
 def main(argv):
     args = parse_args()
     prep_logger(args.log_dir, args.verbose, args.log_stderr, len(args.players))
+    args.players = validate_players(args.players)
     if args.is_tournament:
         run_tournament(args)
     else:
         run_one_game(args)
 
+def validate_players(players):
+    return [player for player in players if locate(player) is not None and
+            isinstance(locate(player)(), HanabiPlayer)]
+
 def run_tournament(args):
     tournament_scores = dict.fromkeys(args.players, 0)
     pairings = list(itertools.combinations(args.players, 2))
+    disqualified = []
     for player1, player2 in pairings:
-        try:
-            game = HanabiGame([player1, player2], args.seed, HanabiVariant(args.variant))
-            game.play_game(args)
-            score = game.table.score()
-        except:
-            #if a player has messed up, penalize both
-            #could diqualify the failing bot if we can determine who failed
-            logger.warning('{player1} or {player2} failed to complete a game'.format(
-                player1 = player1,
-                player2 = player2
-                ))
-            score = -25
-        finally:
-            for handler in logger.handlers:
-                if handler.__class__ is RotatingFileHandler:
-                    handler.doRollover()
-            tournament_scores[player1] += score
-            tournament_scores[player2] += score
+        if player1 and player2 not in disqualified:
+            try:
+                game = HanabiGame([player1, player2], args.seed, HanabiVariant(args.variant))
+                game.play_game(args)
+                score = game.table.score()
+                for handler in logger.handlers:
+                    if handler.__class__ is RotatingFileHandler:
+                        handler.doRollover()
+                tournament_scores[player1] += score
+                tournament_scores[player2] += score
+            except InvalidHanabiMoveException as err:
+                logger.error(err.message)
+                disqualified.append(player1 if err.player_id == 0 else player2)
+                tournament_scores = {k : v for k, v in tournament_scores.iteritems() if k not in disqualified}
+                logger.warning('removed player {player} from tournament'.format(player = disqualified))
     tournament_scores = {k: v / len(pairings) for k, v in tournament_scores.iteritems()}
```

Comment:
#3 appears again. This should be `len(pairings)-len(to_remove)` AT LEAST, since you can't update pairings in the middle of the loop